pyproject.toml

```diff
@@ -0,0 +1,28 @@
+[tool.poetry]
+name = "allowed"
+version = "1.1.0"
+description = "Check if a program only uses a subset of the Python language."
+authors = ["Michel Wermelinger <something@example.com>"]
+readme = "README.md"
+license = "BSD 3-Clause License"
+
+[tool.poetry.dependencies]
+python = "^3.10"
```

Comment:
This is allowing versions of Python >=3.10.0 <4.0.0 anything up to a change in the major version number. When I ran the set up Python 3.11 was used in the environment . I believe something like `"~3.10"` (note the change from `^` to `~`) would only allow patch level changes hence keeping it to 3.10 see  [here](https://python-poetry.org/docs/dependency-specification/) for reference.

Reply:
Thanks for reviewing. I've made the Python version change you suggested. I had a quick look at the `pytype` [requirements](https://github.com/google/pytype#requirements) and it seems that 3.8 to 3.11 is supported, so perhaps there has been an update. I also removed `freezegun` from the tools because I don't think we will need that for this project. Let me know if you spot anything else.

Reply:
Thanks, Yes it looks like `pytype` is now supporting 3.11 as you say.  It might be worth raising an issue about this to discuss further. AFAIK `pytype` was the main reason for the Python 3.10 dependency in this project.